### PR TITLE
Add `model_type_pahm` parameter support

### LINKED ISSUE #61: Add PaHM model parameter to `pyschsim` `BestTrackForcing`
Recently support was added in SCHISM for specifying what model PaHM uses (GAHM vs Symmetric).

### LINKED ISSUE #63: NWM bug: keyword-only argument passed as positional
The following line throws an exception:

https://github.com/schism-dev/pyschism/blob/fb4d23bcf43c95e1717c8aa546cb07c74b34cffa/pyschism/forcing/source_sink/nwm.py#L337

### PR DESCRIPTION
**Pull request by SorooshMani-NOAA**
Fix #63 
Fix #61 

Recently a new parameter was introduced in `param.nml` in [`90a67ff`](https://github.com/schism-dev/schism/commit/90a67ff3747a884bcb48a290a8de6946acabd9fb). This will control what model is used by integrated PaHM code, GAHM: `10` vs Symmetric: `1`

### CODE REVIEW

**Comment by SorooshMani-NOAA** [comment]
@cuill this pull request also includes a bugfix for NWM. The bug (fixed at [`a3b6a5b`](https://github.com/schism-dev/pyschism/pull/62/commits/a3b6a5b53c3acb77034478ac8ca18f79792165b1)) causes problems in automated testing for all packages relying on `pyschism` such as `coupledmodeldriver` and `ensembleperturbation`

File: pyschism/forcing/source_sink/nwm.py [hunk 1]
```diff
@@ -334,7 +334,7 @@ def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
                             if not is_within_directory(path, member_path):
                                 raise Exception("Attempted Path Traversal in Tar File")
                     
-                        tar.extractall(path, members, numeric_owner) 
+                        tar.extractall(path, members, numeric_owner=numeric_owner)
```
Comment (SorooshMani-NOAA) [inline pyschism/forcing/source_sink/nwm.py:337]: `numeric_owner` is a keyword-only argument

**Comment by cuill** [comment]
@SorooshMani-NOAA done

**Comment by yunfangsun** [comment]
@SorooshMani-NOAA 
Hi Soroosh,
Could I know how to add model_type_pahm in the Pyschism script?

Thank you!

**Comment by SorooshMani-NOAA** [comment]
@yunfangsun when creating `BestTrackForcing` you can pass the model type as `hurricane_model` that should be either equal to `"gahm"` or `"symmetric"`. e.g.:
```python
meteo = BestTrackForcing(..., hurricane_model="symmetric", ...)
```

**Comment by yunfangsun** [comment]
Hi @SorooshMani-NOAA 
My Pyschism vision is Version: 0.1.5.
And I get the following error when adding hurricane_model:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[21], line 5
      2 dramp = timedelta(days=10.)
      4 atmospheric = BestTrackForcing.from_nhc_bdeck(nhc_bdeck='hurricane-track.dat') 
----> 5 atmospheric = BestTrackForcing(hurricane_model="symmetric")
      7 #storm_id="Ian2022"
      9 fgrid = pyschism.mesh.ManningsN.linear_with_depth(
     10     hgrid,
     11     min_value=0.02, max_value=0.05,
     12     min_depth=-1.0, max_depth=-3.0)

TypeError: __init__() got an unexpected keyword argument 'hurricane_model'

**Comment by SorooshMani-NOAA** [comment]
@yunfangsun you need to have version `0.1.6` or higher for the `hurricane_model` parameter

**Pull request merged**